src/control/ControlUtils.cpp

```diff
@@ -219,29 +92,46 @@ Vector2 ControlUtils::accelerationLimiter(const Vector2 &targetVel, const Vector
     return prevVel + deltaVel.stretchToLength(finalAcceleration);
 }
 
-/// Get the intersection of two lines
-Vector2 ControlUtils::twoLineIntersection(const Vector2 &a1, const Vector2 &a2, const Vector2 &b1, const Vector2 &b2) {
-    // https://en.wikipedia.org/wiki/Line%E2%80%93line_intersection
-    double denominator = ((a1.x - a2.x) * (b1.y - b2.y) - (a1.y - a2.y) * (b1.x - b2.x));
-    if (denominator != 0) {
-        double numerator = ((a1.x - b1.x) * (b1.y - b2.y) - (a1.y - b1.y) * (b1.x - b2.x));
-        double t = numerator / denominator;
-        return (a1 + (Vector2){t, t} * (a2 - a1));
-    } else {
-        return Vector2();
+/// Calculate the force of a given vector + a certain type.
+/// the basic formula is: force = weight/distance^2 * unit vector
+Vector2 ControlUtils::calculateForce(const Vector2 &vector, double weight, double minDistance) {
+    // if the object is close enough, it's forces should affect. Otherwise don't change anything.
+    if (vector.length() < minDistance && vector.length2() > 0) {
+        return vector.normalize() * (weight / vector.length2());
+    }
+    return {0, 0};
+}
+
+bool ControlUtils::objectVelocityAimedToPoint(const Vector2 &objectPosition, const Vector2 &velocity, const Vector2 &point, double maxDifference) {
+    double exactAngleTowardsPoint = (point - objectPosition).angle();
+
+    // Note: The angles should NOT be constrained here. This is necessary.
+    return (velocity.length() > 0 && velocity.angle() > exactAngleTowardsPoint - maxDifference / 2 && velocity.angle() < exactAngleTowardsPoint + maxDifference / 2);
+}
+
+bool ControlUtils::robotIsAimedAtPoint(int id, bool ourTeam, const Vector2 &point, const world_new::view::WorldDataView world, double maxDifference) {
+    auto robot = world.getRobotForId(id, ourTeam);
+    if (robot) {
+        Angle exactAngleTowardsPoint = (point - (*robot)->getPos());
+        return abs(exactAngleTowardsPoint - (*robot)->getAngle()) < maxDifference;
     }
+    return false;
 }
-/// returns true if the line intersects in the positive extension from point a1 to a2 with the extended line through b1 and b2
-double ControlUtils::twoLineForwardIntersection(const Vector2 &a1, const Vector2 &a2, const Vector2 &b1, const Vector2 &b2) {
-    double denominator = ((a1.x - a2.x) * (b1.y - b2.y) - (a1.y - a2.y) * (b1.x - b2.x));
-    if (denominator != 0) {
-        double numerator = ((a1.x - b1.x) * (b1.y - b2.y) - (a1.y - b1.y) * (b1.x - b2.x));
-        double t = numerator / denominator;
-        return t;
-    } else {
-        return -1.0;
+
+bool ControlUtils::clearLine(const Vector2 &fromPos, const Vector2 &toPos, const world_new::view::WorldDataView world, double safeDistanceFactor, bool includeKeeper) {
+    double minDistance = Constants::ROBOT_RADIUS() * safeDistanceFactor;
```

Comment:
We should not use the old constants anymore but i'm not sure (@RobotJesse ?)

Reply:
Yeah constants should not be used anymore

Reply:
That is the reason why dead code should be removed, because we currently are more busy discussing it than it would take to remove it instead.